crates/net/eth-wire/src/types/message.rs

```diff
@@ -287,6 +289,26 @@ pub trait EncodableExt {
     /// Users of this method should be aware that the truncated data may not represent the complete
     /// original dataset.
     fn encode_truncate(&self, approx: usize, limit: usize) -> Vec<u8>;
+
+    /// Encodes the data up to the specified size limit, returning the encoded data along with
+    /// a flag indicating whether the limit was exceeded. This method is useful for partial
+    /// encoding scenarios where it is necessary to encode as much as possible without going
+    /// over the limit.
```

Comment:
```suggestion
    /// Encodes the data up to the specified size limit, returning the encoded data along with
    /// a flag indicating whether the limit was exceeded.
```

think the use case is obvious from the first sentence :)